tests/test_fields.py

```diff
@@ -492,6 +504,84 @@ def test_no_source_on_child(self):
+        ({'lower': 1., 'upper': 2.},
+         NumericRange(**{'lower': 1, 'upper': 2})),
+        ({'lower': 1},
+         NumericRange(**{'lower': 1})),
+        ({'upper': 1},
+         NumericRange(**{'upper': 1})),
+        ({'empty': True},
+         NumericRange(**{'empty': True})),
+        ({}, NumericRange()),
+    ]
+    invalid_inputs = [
+        ({'lower': 'a'}, ['A valid number is required.']),
+        ('not a dict', ['Expected a dictionary of items but got type "str".']),
+    ]
+    outputs = [
+        (NumericRange(**{'lower': '1.1', 'upper': '2'}),
+         {'lower': '1.1', 'upper': '2', 'bounds': '[)'}),
+        (NumericRange(**{'empty': True}), {'empty': True}),
+        (NumericRange(), {'bounds': '[)', 'lower': None, 'upper': None}),
+    ]
+    field = DecimalRangeField()
+
+    def test_no_source_on_child(self):
+        with pytest.raises(AssertionError) as exc_info:
+            IntegerRangeField(child=serializers.IntegerField(source='other'))
+
+        assert str(exc_info.value) == (
+            "The `source` argument is not meaningful when applied to a `child=` field. "
+            "Remove `source=` from the field declaration."
+        )
+
+
+class TestDecimalRangeFieldWithChildAttribute(FieldValues):
+    serializer_class = DecimalRangeSerializerWithChildAttribute
+    field = DecimalRangeField(child=DecimalField(max_digits=5, decimal_places=2))
+
+    valid_inputs = [
+        ({'lower': '1', 'upper': 2., 'bounds': '[)'},
+         NumericRange(**{'lower': 1., 'upper': 2., 'bounds': '[)'})),
+        ({'lower': 1., 'upper': 2.},
+         NumericRange(**{'lower': 1, 'upper': 2})),
+        ({'lower': 1},
+         NumericRange(**{'lower': 1})),
+        ({'upper': 1},
+         NumericRange(**{'upper': 1})),
+        ({'empty': True},
+         NumericRange(**{'empty': True})),
+        ({}, NumericRange()),
+    ]
+    invalid_inputs = [
+        ({'lower': 'a'}, ['A valid number is required.']),
+        ({'upper': '123456'}, ['Ensure that there are no more than 5 digits in total.']),
+        ({'lower': '9.123'}, ['Ensure that there are no more than 2 decimal places.']),
+        ('not a dict', ['Expected a dictionary of items but got type "str".']),
+    ]
+    outputs = [
+        (NumericRange(**{'lower': '1.1', 'upper': '2'}),
+         {'lower': '1.10', 'upper': '2.00', 'bounds': '[)'}),
+        (NumericRange(**{'empty': True}), {'empty': True}),
+        (NumericRange(), {'bounds': '[)', 'lower': None, 'upper': None}),
+    ]
+
+    def test_no_source_on_child(self):
+        with pytest.raises(AssertionError) as exc_info:
+            IntegerRangeField(child=serializers.IntegerField(source='other'))
+
+        assert str(exc_info.value) == (
+            "The `source` argument is not meaningful when applied to a `child=` field. "
+            "Remove `source=` from the field declaration."
+        )
+
+
 class TestFloatRangeField(FieldValues):
```

Comment:
Shouldn't  `TestFloatRangeField` be run if `compat.FloatRangeField` is not None?
Earlier there was a such a line which is removed recently.
```
@pytest.mark.skipif(django.VERSION < (1, 8) or compat.postgres_fields is None,
                    reason='RangeField is only available for django1.8+'
                    ' and with psycopg2.')
```

I think we need such a decorator to skip test if Floatrange is not supported.